pkg/vcdsdk/ip_spaces_test.go

```diff
@@ -86,15 +87,100 @@ func TestFetchIpSpacesBackingGateway(t *testing.T) {
+	assert.NoError(t, err, "There should be no error parsing ip space test data file content.")
+
+	vcdClient, err := getTestVCDClient(vcdConfig, map[string]interface{}{
+		"user":         testData.Username,
+		"userOrg":      testData.UserOrg,
+		"refreshToken": testData.RefreshToken,
+		"org":          testData.UserOrg,
+		"vdc":          testData.Orgvdc,
+		"getVdcClient": true,
+	})
+	assert.NoError(t, err, "Unable to get VCD client")
+	require.NotNil(t, vcdClient, "VCD Client should not be nil")
+
+	ctx := context.Background()
+
+	gm, err := NewGatewayManager(ctx, vcdClient, testData.OrgvdcNetworkIpSpace, "", testData.Orgvdc)
+	assert.NoErrorf(t, err, "Unable to create Gateway Manager for network %s", testData.OrgvdcNetworkIpSpace)
+
+	fmt.Println("Calling FetchIpSpacesBackingGateway for gateway using Ip Spaces")
+	ipSpaceIds, err := gm.FetchIpSpacesBackingGateway(ctx)
+	assert.NoErrorf(t, err, "Error calling method FetchIpSpacesBackingGateway for netowrk %s", testData.OrgvdcNetworkIpSpace)
+	assert.Greater(t, len(ipSpaceIds), 0)
+
+	fmt.Println("Calling FilterIpSpacesByType for gateway using Ip Spaces")
+	ipSpaces, err := gm.FilterIpSpacesByType(ipSpaceIds, "PUBLIC")
+	assert.NoErrorf(t, err, "Error calling method FilterIpSpacesByType for Ids %s", ipSpaceIds)
+	assert.Equal(t, 1, len(ipSpaces))
+
+	ipSpace := ipSpaces[0]
+
+	fmt.Println("Calling AllocateIpFromIpSpace for gateway using Ip Spaces")
+	allocationId, allocatedIp, err := gm.AllocateIpFromIpSpace(ipSpace)
+	assert.NoError(t, err, "Error calling method AllocateIpFromIpSpace for IpSpace [%s]", ipSpace.IpSpace.Name)
+	assert.NotNil(t, allocationId, "Expected non nil Allocation Id")
+	assert.NotNil(t, allocatedIp, "Expected non nil Allocated Ip")
+
+	fmt.Println("Calling FindIpAllocationByIp for gateway using Ip Spaces")
+	retrievedIpSpaceAllocation, err := gm.FindIpAllocationByIp(ipSpace, allocatedIp)
+	assert.NoError(t, err, "Error calling method FindIpAllocationByIp for IpSpace [%s], Ip [%s]", ipSpace.IpSpace.Name, allocatedIp)
+	assert.NotNil(t, retrievedIpSpaceAllocation)
+	assert.Equal(t, allocationId, retrievedIpSpaceAllocation.IpSpaceIpAllocation.ID, "")
+	assert.Equal(t, allocatedIp, retrievedIpSpaceAllocation.IpSpaceIpAllocation.Value, "")
+
+	dummyMarker := "Dummy Marker"
+	fmt.Println("Calling MarkIpAsUsed for gateway using Ip Spaces")
+	updatedIpSpaceAllocation, err := gm.MarkIpAsUsed(retrievedIpSpaceAllocation, dummyMarker)
+	assert.NoError(t, err, "Error calling method MarkIpAsUsed for Allocation [%s]", retrievedIpSpaceAllocation.IpSpaceIpAllocation.ID)
+	assert.NotNil(t, updatedIpSpaceAllocation)
+	assert.Equal(t, types.IpSpaceIpAllocationUsedManual, updatedIpSpaceAllocation.IpSpaceIpAllocation.UsageState)
+	assert.Equal(t, dummyMarker, updatedIpSpaceAllocation.IpSpaceIpAllocation.Description)
+
+	fmt.Println("Calling FindIpAllocationByMarker for gateway using Ip Spaces")
+	retrievedIpSpaceAllocation2, err := gm.FindIpAllocationByMarker(ipSpace, dummyMarker)
```

Comment:
Nit: It will be helpful to have a test that uses a 'Random Marker' and gets the nil IP Allocation.

Reply:
Added the test.